src/plugins/data/common/expressions/interpreter_provider.ts

```diff
@@ -71,6 +71,11 @@ export function interpreterProvider(config: any) {
       // if something failed, just return the failure
       if (getType(newContext) === 'error') return newContext;
 
+      // if execution was aborted return error
+      if (handlers.abortSignal && handlers.abortSignal.aborted) {
+        return createError(`abort was requested`);
```

Comment:
`createError` is surfaced in the UI... is this definitely the behavior we want? Like, if we add a button for someone to cancel, do we want to always see an error message, since technically this would be the expected behavior from a user perspective? (Or maybe it should be a confirmation/success message?)

Also, I'm wondering if it would be better to `throw` an error here instead, and catch it below? That way all of our error handling happens in one place.

Reply:
Agreed, I've changed it to a simple `Error` in https://github.com/elastic/kibana/pull/40238/commits/0c429c1aa59523267b2ae07185a225e8adaf2276. We should probably eventually make a separate `AbortedError` or something of the sort.